python/src/scipp/plotting/model.py

```diff
@@ -113,6 +113,13 @@ def _axis_coord(self, array, dim):
                 coord = arange(dim=dim, start=0, stop=array.sizes[dim])
             elif typing.has_datetime_type(coord):
                 coord = coord - coord.min()
+            # TODO This hack looks "wrong". This may be an indicator that resampling
+            # may not be the correct default choice.
+            # If there is a bin-edge coord but no corresponding event-coord then `bin`
+            # cannot handle this. We could first bin without this and then use `rebin`.
```

Comment:
Possibly, but it would run into memory issues in general if the axis to be rebinned is large. I think this is something that will feed into the discussion started in #2167.